include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_merge_sort.h

```diff
@@ -159,6 +306,51 @@ struct __parallel_sort_submitter<_IdType, __internal::__optional_kernel_name<_Le
     }
 };
 
+template <typename _Range, typename _Compare>
+struct __leaf_sorter_selector
+{
+    // 1024 is greater than or equal the maximum work-group size for the majority of devices
+    // 8 is the maximum reasonable value for bubble sub-group sorter
+    using _LeafXL = __leaf_sorter<8, 1024, _Range, _Compare>;
+    using _LeafL = __leaf_sorter<4, 512, _Range, _Compare>;
+    using _LeafM = __leaf_sorter<2, 256, _Range, _Compare>;
+    // 2 is the smallest reasonable value for merge-path group sorter
+    // SYCL specification requires that local memory size should be at least 32 KB,
+    // what is enough to sort 256 byte elements (32KB / (2 * 64)).
+    // It is unlikely that the element to sort is larger than 256 bytes.
+    using _LeafS = __leaf_sorter<2, 64, _Range, _Compare>;
+
+    using _Tp = oneapi::dpl::__internal::__value_t<_Range>;
+
+    std::variant<_LeafXL, _LeafL, _LeafM, _LeafS>
+    select(const sycl::queue& __q, _Range& __rng, _Compare __comp) const
+    {
+        const auto __max_wg_size = __q.get_device().template get_info<sycl::info::device::max_work_group_size>();
+        auto __max_slm_items = __q.get_device().template get_info<sycl::info::device::local_mem_size>() / sizeof(_Tp);
+        // Get the work group size adjusted to the local memory limit.
+        // Pessimistically reduce it by 0.8 to take into account memory used by compiled kernel.
```

Comment:
Is .8 enough since we use .5 in similar places?

Reply:
I stuck to something larger than 0.5, because now group algorithms are used. 0.8 was based on "gut feeling".
However, I've replaced it with 0.5, because this is what was agreed upon (I do not know the reason, though). it can be reconsidered later, if needed.

Reply:
I think .5 was very pessimistically since we've seen issues with debug builds. .8 might be fine for this kernel though.